File: app/src/main/java/com/nimtego/plectrum/data/network/wiki/WikiApi.java
package com.nimtego.plectrum.data.network.wiki;

import com.nimtego.plectrum.data.model.wiki.WikiSearchResult;

import java.util.Map;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

public interface WikiApi {
    @GET("api.php")
    Observable<WikiSearchResult> searchArtist(@QueryMap Map<String, String> param);
}

File: app/src/main/java/com/nimtego/plectrum/data/network/wiki/WikiApi.kt
package com.nimtego.plectrum.data.network.wiki

import com.nimtego.plectrum.data.model.wiki.WikiSearchResult

import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.QueryMap

interface WikiApi {
    @GET("api.php")
    fun searchArtist(@QueryMap param: Map<String, String>): Observable<WikiSearchResult>
}
